fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -17,166 +16,320 @@ import androidx.compose.material.IconButton
 import androidx.compose.material.LocalContentColor
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.CompositionLocalProvider
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
 import mozilla.components.browser.menu.view.MenuButton
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.browser.state.selector.privateTabs
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsState
 import org.mozilla.fenix.R
-import org.mozilla.fenix.components.toolbar.ItemType.STANDARD
-import org.mozilla.fenix.components.toolbar.ItemType.TAB_COUNTER
+import org.mozilla.fenix.compose.LongPressIconButton
 import org.mozilla.fenix.compose.TabCounter
 import org.mozilla.fenix.compose.annotation.LightDarkPreview
 import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.theme.Theme
+import org.mozilla.fenix.theme.ThemeManager
 
 /**
  * Top-level UI for displaying the navigation bar.
  *
- * @param actionItems A list of [ActionItem] used to populate the bar.
- * @param tabCount The number of opened tabs.
- * @param menuButton A [MenuButton] to be used for [ItemType.MENU].
+ * @param onBackButtonClick
+ * @param onBackButtonLongPress
+ * @param onForwardButtonClick
+ * @param onForwardButtonLongPress
+ * @param onHomeButtonClick
+ * @param menuButton
+ * @param onTabsButtonClick
+ * @param isPrivateMode
+ * @param browserStore
  */
+@Suppress("LongParameterList")
 @Composable
-fun NavigationBar(
-    actionItems: List<ActionItem>,
-    tabCount: Int,
-    menuButton: MenuButton? = null,
+fun OpenTabNavBar(
+    onBackButtonClick: () -> Unit,
+    onBackButtonLongPress: () -> Unit,
+    onForwardButtonClick: () -> Unit,
+    onForwardButtonLongPress: () -> Unit,
+    onHomeButtonClick: () -> Unit,
+    menuButton: MenuButton,
+    onTabsButtonClick: () -> Unit,
+    isPrivateMode: Boolean,
+    browserStore: BrowserStore,
```

Comment:
nit: make sure to re-order these parameters (non-optional variables -> optional variables -> lambdas)